File: sentimental-mario-more/mario.py
def main():
    while True:
        try:
            height = int(input("Height: "))
            if 1 <= height <= 8:
                grid(height)
                break
            else:
                print("Please enter a number between 1 and 8.")
        except ValueError:
            print("Invalid input. Please enter a number.")

def grid(height):
    for i in range(1, height + 1):
        # Print leading spaces for left-aligned pyramid
        print(" " * (height - i), end="")

        # Print left-aligned pyramid
        print("#" * i, end="")

        # Print space between pyramids
        print("  ", end="")

        # Print right-aligned pyramid
        print("#" * i)

if __name__ == "__main__":
    main()

File: mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int height;

    // Prompt user for a height between 1 and 8
    do
    {
        height = get_int("Height: ");
    } while (height < 1 || height > 8);

    // Loop to print each row of the pyramid
    for (int i = 0; i < height; i++)
    {
        // Print leading spaces
        for (int j = height - i - 1; j > 0; j--)
        {
            printf(" ");
        }

        // Print left pyramid
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }

        // Print gap between pyramids
        printf("  ");

        // Print right pyramid
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }

        // Move to the next line after each row
        printf("\n");
    }

}
